notification/integration-test/src/notification-execution.test.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-env jest */
+const request = require('supertest')
+const waitOn = require('wait-on')
+
+const URL = process.env.NOTIFICATION_API || 'http://localhost:8080'
+
+const MOCK_RECEIVER_PORT = process.env.MOCK_RECEIVER_PORT || 8081
+const MOCK_RECEIVER_HOST = process.env.MOCK_RECEIVER_HOST || 'localhost'
+const MOCK_RECEIVER_URL = 'http://' + MOCK_RECEIVER_HOST + ':' + MOCK_RECEIVER_PORT
+
+describe('Notification', () => {
+    console.log('Notification-Service URL= ' + URL)
+
+    beforeAll(async() => {
+        const pingUrl = URL + '/'
+        console.log('Waiting for notification-service with URL: ' + pingUrl)
+        console.log('Waiting for mock webhook receiver with URL: ' + MOCK_RECEIVER_URL)
+        await waitOn({ resources: [pingUrl, MOCK_RECEIVER_URL], timeout: 50000 })
+    }, 60000)
+
+    test('Trigger webhook', async() => {
+      // SETUP: store notification config
+      const webhookConfig = {
+          pipelineId: 1,
+          condition: 'data.value1 > 0',
+          url: MOCK_RECEIVER_URL + '/webhook1'
+      }
+      const notificationResponse = await request(URL)
+          .post('/config/webhook')
+          .send(webhookConfig)
+      expect(notificationResponse.status).toEqual(200)
+      const id = notificationResponse.body.id
+
+      // SETUP: trigger event
+      const dataLocation = 'storage/1234'
+      const triggerEvent = {
+          pipelineId: 1,
+          pipelineName: "Integration-Test Pipeline 1",
+          dataLocation: dataLocation,
+          jobResult: {
+            data: {
+              value1: 1
+            },
+            stats: {
+              durationInMilliSeconds: 500,
+              startTimestamp: 12345,
+              endTimestamp: 12345
+            }
+          }
+      }
+
+      // ACT
+      const notificationTriggerResponse = await request(URL)
+            .post('/trigger')
+            .send(triggerEvent)
+
+      expect(notificationTriggerResponse.status).toEqual(200)
+      await sleep(3000) // wait for processing
```

Comment:
Instead of sleeping, we could poll for the notification to arrive like it is done in the system-test

Reply:
Yeah, good point. I'll open an issue for it